integrations/clickup/main.py

```diff
@@ -0,0 +1,170 @@
+        yield teams
+
+
+@ocean.on_resync(ObjectKind.SPACE)
+async def on_resync_spaces(kind: str) -> ASYNC_GENERATOR_RESYNC_TYPE:
+    clickup_client = await init_client()
+    async for spaces in clickup_client.get_all_spaces():
+        logger.info(f"Received batch of {len(spaces)} spaces")
+        yield spaces
+
+
+@ocean.on_resync(ObjectKind.PROJECT)
+async def on_resync_projects(kind: str) -> ASYNC_GENERATOR_RESYNC_TYPE:
+    clickup_client = await init_client()
+    async for projects in clickup_client.get_all_projects():
+        logger.info(f"Received batch of  {len(projects)} projects")
+        yield projects
+
+
+@ocean.on_resync(ObjectKind.ISSUE)
+async def on_resync_issues(kind: str) -> ASYNC_GENERATOR_RESYNC_TYPE:
+    clickup_client = await init_client()
+    async for issues in clickup_client.get_paginated_issues():
+        logger.info(f"Received batch of {len(issues)} issues")
+        yield issues
+
+
+async def handle_register(
+    clickup_client: Any,
+    entity_id: str,
+    kind: ObjectKind,
+    event_type: str,
+    webhook_id: str,
+) -> None:
+    team_id = WEBHOOK_TEAM_MAP.get(webhook_id)
+    if kind == ObjectKind.ISSUE:
+        entity = await clickup_client.get_single_issue(entity_id)
+    elif kind == ObjectKind.PROJECT:
+        entity = await clickup_client.get_single_project(entity_id, team_id)
+    else:
+        entity = await clickup_client.get_single_space(entity_id, team_id)
+
+    if entity:
+        await ocean.register_raw(kind, [entity])
+        logger.info(f"Registered {kind} for event {event_type}")
+    else:
+        logger.error(f"Handler returned None for entity_id {entity_id}")
+
+
+async def handle_unregister(entity_id: str, kind: ObjectKind, event_type: str) -> None:
+    try:
+        await ocean.unregister_raw(kind, [{"id": entity_id}])
+        logger.info(f"Unregistered {kind} for event {event_type}")
+    except Exception as e:
+        logger.error(f"Exception {e} occurred while attempting to unregister raw")
+
+
+@ocean.router.post("/webhook")
+async def handle_webhook_request(data: Dict[str, Any]) -> Dict[str, Any]:
+    """
+    Handle the webhook request from ClickUp.
+    Events are mapped to the appropriate actions in event_handlers.
+    """
+    clickup_client = await init_client()
+    logger.info(f"Received webhook event of type: {data.get('event')}")
+
+    event_handlers = {
+        "task": ObjectKind.ISSUE,
+        "list": ObjectKind.PROJECT,
+        "space": ObjectKind.SPACE,
+    }
+
+    event_type = data["event"]
+    webhook_id = data["webhook_id"]
+    action = "unregister" if "Deleted" in event_type else "register"
+
+    for key, kind in event_handlers.items():
+        if key in event_type:
+            entity_id = data.get(f"{key}_id")
+            if not entity_id:
+                logger.error(f"No {key}_id found in data for event {event_type}")
+                continue
+            if action == "register":
+                await handle_register(
```

Comment:
if `action` is only used ones, I feel like it is redundant to have that assigned to an attribute, rather then just having that logic in the if statement

Reply:
e.g. 

```
if `Deleted` in event_type`: 
   await handle_register(....
```